### PR TITLE
Add Twitter DNT meta tag.

### PR DESCRIPTION
**Pull request by cowlicks**
Twitter recently withdrew support for allowing their users to opt out of tracking with DNT. But they still quietly support this privacy policy in the case where website publisher's choose to use it when they embed Twitter content. They no longer call this privacy policy "DNT", but they have confirmed that the policy has not changed.

This PR embeds a meta tag on websites that load content from twitter. Twitter has no* way of distinguishing meta tags embedded by the website publisher, or by an extension in the user's browser. So they should continue to apply their DNT policy on visited with PB.

Twitter's documentation about this privacy policy is here: https://dev.twitter.com/web/overview/privacy#what-privacy-options-do-website-publishers-have

(*) no practical method to my knowledge

### CODE REVIEW

**Comment by cowlicks** [comment]
@jsha just pinging you to check if what I've said here sounds right. No need to review code.

**Comment by jsha** [comment]
Text sounds good. There are probably some tricky timing issues around whether this fires before widgets.js checks for it.

**Comment by koops76** [comment]
@jsha Maybe inject it directly into the server's response?

**Comment by koops76** [comment]
https://twitter.com/personalization

This page is Twitter's replacement of their removed DNT setting. I think we can try setting the same cookies it does.

**Comment by cowlicks** [comment]
@koops76 @jsha The pull request as is does not guarantee that, but it is straight forward to do so by hooking this in to webrequests. I just wanted to put this out so I could talk about it in a blog post.

**Comment by cowlicks** [comment]
This is really an abuse of our widgets system, but  I think refactoring that is better done as a separate pull request.

File: src/js/webrequest.js [hunk 1]
```diff
@@ -566,9 +566,9 @@ function _isTabChromeInternal(tabId) {
  */
 function _isTabAnExtension(tabId) {
   return (
-    _frameUrlStartsWith(tabId, "chrome-extension://") ||
-    _frameUrlStartsWith(tabId, "moz-extension://")
-   );
+   _frameUrlStartsWith(tabId, "chrome-extension://") ||
+   _frameUrlStartsWith(tabId, "moz-extension://")
+  );
```
Comment (ghostwords) [inline src/js/webrequest.js]: This is a big-time nit, buuuut, removing the space from the closing parenthesis is great, that was a typo, but creating single-space indent above isn't so great; an indentation level should be two spaces.

**Comment by koops76** [comment]
@ghostwords 👍

**Comment by cowlicks** [comment]
Always injecting would make it very easy to fingerprint Privacybadger users. Sites can already do this, however with the suggested change, it would make it trivial.

Another issue is that such a content script would *always* be run, even if privacybadger was deactivated for the site.

I see the widget replacement system eventually becoming more a more general purpose system for running code in webpages to modify them. For example unwrapping url's could be part of it eventually.

**Comment by ghostwords** [comment]
Can you explain how exactly "always injecting would make it very easy to fingerprint Privacybadger users"? I don't see how adding a new content script (that does the same work this PR's changes to socialwidgets.js do) make Privacy Badger more identifiable.

>Another issue is that such a content script would always be run, even if privacybadger was deactivated for the site.

This is an existing issue with anything race-condition sensitive. You either guarantee on-time injection, or you get conditional injection, but not both. The fix for this is tracked in https://crbug.com/478183. Injecting unconditionally seems preferable to working some of the time.

**Pull request closed without merging**